examples/README.md

```diff
@@ -0,0 +1,279 @@
+        LOG_DEBUG("HelloworldCC: Response buffer too small");
+        *actual_response_len = 0;
+        return -1;
+    }
+
+    // copy result to response
+    memcpy(response, result.c_str(), neededSize);
+    *actual_response_len = neededSize;
+    LOG_DEBUG("HelloworldCC: Response: %s", result.c_str());
+    LOG_DEBUG("HelloworldCC: +++ Executing done +++");
+    return 0;
+}
+
+//  Add asset_name, value to ledger
+std::string putData(std::string asset_name, int value, void*ctx)
+{
+    LOG_DEBUG("HelloworldCC: +++ putData +++");
+
+    put_state(asset_name.c_str(), (uint8_t*)&value, sizeof(int), ctx);
+
+    return OK;
+}
+
+//  Get value set for asset_name in ledger 
+std::string getData(std::string asset_name, void*ctx)
+{
+    std::string result;
+    LOG_DEBUG("HelloworldCC: +++ getData +++");
+
+    uint32_t asset_bytes_len = 0;
+    uint8_t asset_bytes[MAX_VALUE_SIZE];
+    get_state(asset_name.c_str(), asset_bytes, sizeof(asset_bytes), &asset_bytes_len, ctx);
+
+    //  check if asset_name exists
+    if (asset_bytes_len > 0)
+    {
+        //  asset exists;  return value
+        result = marshall (asset_name, *asset_bytes);
+    }
+    else
+    {
+        //  asset does not exist
+        result = NOT_FOUND;
+    }
+    return result;
+}
+
+std::string marshall (std::string asset_name, uint8_t value)
+{
+    JSON_Value* root_value = json_value_init_object();
+    JSON_Object* root_object = json_value_get_object(root_value);
+    json_object_set_string(root_object, "name", asset_name.c_str());
+    json_object_set_number(root_object, "value", value);
+    char* serialized_string = json_serialize_to_string(root_value);
+    std::string out(serialized_string);
+    json_free_serialized_string(serialized_string);
+    json_value_free(root_value);
+    return out;
+}
+
+```
+### Build 
+To build the chaincode, add `CMakefile.txt` and `Makefile` (Note that CMakeLists.txt is a modified version of examples/auction/CMakeLists.txt.  Makefile is the same as in examples/auction/Makefile).
+File: CMakeLists.txt
+```
+cmake_minimum_required(VERSION 3.5.1)
+
+set(SOURCE_FILES
+    helloworld_cc.cpp
+    )
+
+include(../../ecc_enclave/enclave/CMakeLists-common-app-enclave.txt)
+```
+
+
+Make sure you have the environment variables set.  Now, execute:
+
+
+```make```
+
+
+If you dont see any errors, you are good to go.
+
+Optional: For this `helloworld` chaincode to be included in the build framework, modify the file: `FPC-INSTALL-DIR/examples/Makefile` to include the `helloworld` chaincode.
+
+File: Makefile   
+```TOP = ..
+include $(TOP)/build.mk
+
+EXAMPLES = auction echo helloworld
+
+build clean:
+	$(foreach DIR, $(EXAMPLES), $(MAKE) -C $(DIR) $@ || exit;)
+```
+  
+Next step is to install, instantiate the chaincode and invoke transactions.  For this, 
+you use the integration framework.  Move to `FPC-INSTALL-DIR/integration`.   In the existing `Makefile`,  make these modifications: 
+
```

Comment:
Why would the user want to add this to the integration test? Actually it would be nice if the user could just create the helloworld_test.sh in its example/halloworld folder and run it also from there. So everything is self contained. WDYT?

Reply:
Upon this suggestion, I have modified the tutorial to create a  helloworld_test.sh  in example/helloworld folder.  However, it uses config folder (and the yaml files) in integration folder.